### PR TITLE
fix a config issue with s3 and also fix file permissions

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
I just realized that I have never tested this with minio, I put too much trust in the API. Anyway this is a quick fix to make it work with it. Since we are here, I updated the file permissions per [MM-56973](https://mattermost.atlassian.net/browse/MM-56973)

### CODE REVIEW

**Review by agnivade — APPROVED** [review]

File: server/copy.go [hunk 1]
```diff
@@ -97,7 +101,7 @@ func copyFromFileStore(dst, src string, b filestore.FileBackend) error {
 		if err != nil {
 			return err
 		}
-		f, err := os.Create(fileDest)
+		f, err := os.OpenFile(fileDest, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0660)
```
Comment (esarafianou) [inline server/copy.go:104]: I think this change has no effect as `os.Create(fileDest)` already sets 0660 as permissions. What we had discussed is that only the mattermost process should need to have access and we can restrict the group permissions

File: server/copy.go [hunk 2]
```diff
@@ -134,7 +138,7 @@ func copyFromFileStore(dst, src string, b filestore.FileBackend) error {
 
 func compressDirectory(sourceDir, compressedFile string) error {
 	// Create a new archive file
-	newZipFile, cErr := os.Create(compressedFile)
+	newZipFile, cErr := os.OpenFile(compressedFile, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0750)
```
Comment (esarafianou) [inline server/copy.go:141]: os.Create gives 0660 permissions. 

Why are we changing it giving the user and the group execute permissions? afaik there is nothing to execute. 

Similar to the above, we'd like to restrict the group as only the mattermost process should need to have access

**Review by esarafianou — CHANGES_REQUESTED** [review]

File: server/copy.go (hunk 1 above)
Comment (isacikgoz) [reply server/copy.go:104]: Ah sorry I forgot to change mode 🤦

**Review by esarafianou — APPROVED** [review]

File: server/copy.go (hunk 2 above)
Comment (isacikgoz) [reply server/copy.go]: Right, I probably misinterpreted what we discussed. Changed to more restrictive, please check.

**Pull request merged**